sklearn/feature_selection/tests/test_feature_select.py

```diff
@@ -620,7 +618,8 @@ def test_f_classif_constant_feature():
 
     X, y = make_classification(n_samples=10, n_features=5)
     X[:, 0] = 2.0
-    assert_warns(UserWarning, f_classif, X, y)
+    with ignore_warnings(category=UserWarning):
+        f_classif(X, y)
```

Comment:
We don't want to ignore the warning but assert that a UserWarning is raised instead. The pytest context manager is the right way to do it here as well.
```
with pytest.warns(UserWarning):
    f_classif(X, y)
```

Reply:
Thank you for advising, I fixed 4ca0c04 .